src/MCPClient/lib/clientScripts/verify_and_restructure_transfer_bag.py

```diff
@@ -55,6 +57,20 @@ def restructureBagForComplianceFileUUIDsAssigned(
         dirPath = os.path.join(unitPath, dir)
         dirDataPath = os.path.join(unitPath, "data", dir)
         if os.path.isdir(dirDataPath):
+            # When moving metadata directory from the bag, check if
+            # there is already a metadata dir at the top level
+            # (it could have been created by automation tools)
+            # (ref. AM issue 1022 )
+            topleveldir_exists = False
+            if dir == "metadata" and os.path.isdir(dirPath):
+                # temporarily rename top level metadata directory
+                job.pyprint(
+                    "temporarily renaming pre-existing directory {}".format(dirPath)
+                )
+                topleveldir_exists = True
+                tempdirPath = os.path.join(unitPath, "{}_{}".format(dir, os.getpid()))
+                os.rename(dirPath, tempdirPath)
```

Comment:
```suggestion
                move_or_merge(dirPath, dirDataPath)
```
I wonder what you think of this? (remove lines 67->72 and add this one line here instead). We can then avoid the rename here and then remove lines `87->94` too. Do the results match what you expect still? (for me they look good).

This microservice is quite old and pretty complex, I'm not a fan of the `fileOperations.updateDirectoryLocation(...)` in the next few lines, there's another move going on there (as you know). I'd prefer to keep that all within a single microservice script, and maybe only make calls out to update the DB. But this isn't really in scope to fix here. What I think we can do is avoid adding too much more complexity, so if we can reduce the operations involved from a handful of lines to one, and possibly include some logging, then as a fix, that probably works quite well for the client here.

I've shown an example in a gist [here](https://gist.github.com/ross-spencer/5e4bf0290990b44bed0f4243a2ce7fc3/revisions) to make it clearer. Let me know if that change would be any good.

Reply:
@ross-spencer thank you very much! Using `move_or_merge()` from the beginning makes the logic simpler (like you, also not a fan of `fileOperations.updateDirectoryLocation(...)`, I was afraid that after the change something in the code inside it could produce some issue/error, but a quick test of the fix in the client branch worked good)